src/theme/palette.ts

```diff
@@ -64,3 +65,76 @@ export type Palette = {
 } & Record<typeof mainTypes[number], generatedColorShades>;
 
 export type formFieldStyles = 'filled' | 'outlined' | 'elevated';
+
+/**
+ * this function picks either white or black color based on the background that is passed
+ * swatches are calculated based on accessibility by the design team and splited to those two colors
+ **/
+export const pickTextColorFromSwatches = (
+  color: typeof flatColors[number],
+  shade: typeof colorShades[number]
+) => {
+  const colorsForWhiteText: Partial<Record<
+    typeof colorShades[number],
+    typeof flatColors[number][]
+  >> = {
+    700: [
```

Comment:
I thought that as well but based on the designs and the rule applied if something is added doesnt mean that it will be white for example. This depends on the viewability of the color. So i thought that if i applied something like that would result to an error possibly. 

Also filter for cases like 400 shade would be long. what do you think? If filter is simple i dont have a problem but i am not sure.

Reply:
Hmm I didn't think about that, so I agree with you.